cns/middlewares/mock/mockSWIFTv2.go

```diff
@@ -0,0 +1,206 @@
+	if !ok {
+		return cns.PodIpInfo{}, errFailedToGetPod
+	}
+
+	// Check if the MTPNC CRD is ready. If one of the fields is empty, return error
+	if mtpnc.Status.PrimaryIP == "" || mtpnc.Status.MacAddress == "" || mtpnc.Status.NCID == "" || mtpnc.Status.GatewayIP == "" {
+		return cns.PodIpInfo{}, errMTPNCNotReady
+	}
+	podIPInfo := cns.PodIpInfo{}
+	podIPInfo.PodIPConfig = cns.IPSubnet{
+		IPAddress: mtpnc.Status.PrimaryIP,
+	}
+	podIPInfo.MacAddress = mtpnc.Status.MacAddress
+	podIPInfo.NICType = cns.DelegatedVMNIC
+	podIPInfo.SkipDefaultRoutes = false
+
+	return podIPInfo, nil
+}
+
+func (m *MockSWIFTv2Middleware) SetRoutes(podIPInfo *cns.PodIpInfo) error {
+	podIPInfo.Routes = []cns.Route{}
+	switch podIPInfo.NICType {
+	case cns.DelegatedVMNIC:
+		// default route via SWIFT v2 interface
+		route := cns.Route{
+			IPAddress: "0.0.0.0/0",
+		}
+		podIPInfo.Routes = []cns.Route{route}
+	case cns.InfraNIC:
+		podCIDRs, err := configuration.PodCIDRs()
+		if err != nil {
+			return fmt.Errorf("failed to get podCIDRs from env : %w", err)
+		}
+		podCIDRsV4, podCIDRv6, err := parseCIDRs(podCIDRs)
+		if err != nil {
+			return fmt.Errorf("failed to parse podCIDRs : %w", err)
+		}
+
+		serviceCIDRs, err := configuration.ServiceCIDRs()
+		if err != nil {
+			return fmt.Errorf("failed to get serviceCIDRs from env : %w", err)
+		}
+		serviceCIDRsV4, serviceCIDRsV6, err := parseCIDRs(serviceCIDRs)
+		if err != nil {
+			return fmt.Errorf("failed to parse serviceCIDRs : %w", err)
+		}
+		// Check if the podIPInfo is IPv4 or IPv6
+		if net.ParseIP(podIPInfo.PodIPConfig.IPAddress).To4() != nil {
+			// routes for IPv4 podCIDR traffic
+			for _, podCIDRv4 := range podCIDRsV4 {
+				podCIDRv4Route := cns.Route{
+					IPAddress:        podCIDRv4,
+					GatewayIPAddress: overlayGatewayv4,
+				}
+				podIPInfo.Routes = append(podIPInfo.Routes, podCIDRv4Route)
+			}
+			// route for IPv4 serviceCIDR traffic
+			for _, serviceCIDRv4 := range serviceCIDRsV4 {
+				serviceCIDRv4Route := cns.Route{
+					IPAddress:        serviceCIDRv4,
+					GatewayIPAddress: overlayGatewayv4,
+				}
+				podIPInfo.Routes = append(podIPInfo.Routes, serviceCIDRv4Route)
+			}
+
+		} else {
+			// routes for IPv6 podCIDR traffic
+			for _, podCIDRv6 := range podCIDRv6 {
+				podCIDRv6Route := cns.Route{
+					IPAddress:        podCIDRv6,
+					GatewayIPAddress: overlayGatewayV6,
+				}
+				podIPInfo.Routes = append(podIPInfo.Routes, podCIDRv6Route)
+			}
+			// route for IPv6 serviceCIDR traffic
+			for _, serviceCIDRv6 := range serviceCIDRsV6 {
+				serviceCIDRv6Route := cns.Route{
+					IPAddress:        serviceCIDRv6,
+					GatewayIPAddress: overlayGatewayV6,
+				}
+				podIPInfo.Routes = append(podIPInfo.Routes, serviceCIDRv6Route)
+			}
+		}
+	default:
+		return errInvalidSWIFTv2NICType
+	}
+	return nil
+}
+
+// parseCIDRs parses the semicolons separated CIDRs string and returns the IPv4 and IPv6 CIDRs.
+func parseCIDRs(cidrs string) (v4IPs, v6IPs []string, err error) {
```

Comment:
This looks like the same method in original file as well. Can we put this in a helper file or reuse existing ones to avoid duplicates?